File: js/runner.js
const fs = require("fs");
const path = require("path");

const {Item, update_quality} = require('./src/gilded_rose');

if (process.argv.length < 3) {
  console.warn("Usage: node runner.js <in-file> <out-file>");
  process.exit(1);
}

const inFile = process.argv[2];
const outFile = process.argv[3];

console.warn("Reading from %s, writing to %s", inFile, outFile);

function parseItem(line) {
  var [name, sell_in, quality] = line.split("__");
  return new Item(name, Number(sell_in), Number(quality));
}
function unparseItem(item) {
  return [item.name, item.sell_in, item.quality].join("__");
}

const items = fs.readFileSync(inFile, "utf8")
  .split("\n")
  .map(l => l.trim())
  .filter(l => l && !l.startsWith(";"))
  .map(parseItem);

const output = update_quality(items)
  .map(unparseItem)
  .join("\n");

fs.mkdirSync(path.dirname(outFile), {recursive: true});
fs.writeFileSync(outFile, output);

File: ruby/runner.rb
#!/usr/bin/ruby -w

require 'fileutils'

require File.join(File.dirname(__FILE__), 'gilded_rose')

if ARGV.length != 2
  puts "Usage: ruby runner.rb <in-file> <out-file>"
  exit 1
end

in_file, out_file = ARGV

puts "Reading from #{in_file}, writing to #{out_file}"

items = IO.readlines(in_file)
  .map(&:strip)
  .reject { |l| l.empty? || l.start_with?(";") }
  .map { |l| l.split("__") }
  .map { |name, sell_in, quality|
    Item.new(name, sell_in.to_i, quality.to_i)
  }

rose = GildedRose.new(items)
rose.update_quality

output = items
  .map { |i| [i.name, i.sell_in, i.quality].join("__") }
  .join("\n")

FileUtils.mkdir_p(File.dirname(out_file))
IO.write(out_file, output)
